### DISCUSSION TITLE
Query template and Refinement filters

### DISCUSSION THREAD

**Discussion by schoier** [microsoft-search/pnp-modern-search#3307] [General]
What is the difference between Query template and Refinement filters? It might be a general SharePoint Search question, but I have tried googling it without finding any satisfying answers?

![image](https://github.com/microsoft-search/pnp-modern-search/assets/33417064/7447ea72-4d10-4932-838a-22e786df3e8c)

**Comment by kasperbolarsen** [comment]
I think the best explanation is this one: https://learn.microsoft.com/en-us/sharepoint/dev/general-development/query-refinement-in-sharepoint
To be honest I rarely use the Refinement filters for anything.

**Comment by wobba** [comment]
The main difference is that query template takes KQL while the refinement filters takes FQL. Very overlapping, but FQL has different notation and support some more options which may or may not be of use. And a hidden pro-tip is that refinement filters do not have a max length, while KQL has.
